src/rgw/rgw_kafka.cc

```diff
@@ -416,13 +418,10 @@ class Manager {
       for (;conn_it != end_it;) {
         
         auto& conn = conn_it->second;
-        // delete the connection if marked for deletion
-        if (conn->marked_for_deletion) {
-          ldout(conn->cct, 10) << "Kafka run: connection is deleted" << dendl;
-          conn->destroy(STATUS_CONNECTION_CLOSED);
-          std::lock_guard lock(connections_lock);
-          // erase is safe - does not invalidate any other iterator
-          // lock so no insertion happens at the same time
+
+        // Checking the connection idlesness
+        if(conn->timestamp.sec() + 30 < ceph_clock_now()) {
```

Comment:
please add a data member to the Manager called "max_idle_time" and set it to 30 in the ctor (similarly to the "max_connection", see lines 295-297)